### DISCUSSION TITLE
部署在 netlify 使用 gemini 会被打断？

### DISCUSSION THREAD

**Discussion by xingpingcn** [lobehub/lobehub#2655] [Self Hosting] [answered]
### 💻 Operating System

Windows

### 📦 Environment

Other

### 🌐 Browser

Chrome

### 🐛 Bug Description

部署在netlify使用gemini会莫名其妙地自行打断，但是使用官网提供的preview就没有这种情况

### 📷 Recurrence Steps

复现条件不清楚，有些时候输出几个字就自己打断，有时候输出100字就自己打断。

[a comment by an automated account is not shown]

**Comment by sxjeru** [comment]
vercel 托管也有类似情况。

[a comment by an automated account is not shown]

**Comment by xingpingcn** [comment]
> vercel 托管也有类似情况。

那现在是只能用自己的vps用docker托管比较靠谱？

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
> 部署在netlify使用gemini会莫名其妙地自行打断

在控制台有没有抛错？

> 官网提供的preview就没有这种情况

preview 用的是 vercel

[a comment by an automated account is not shown]

**Comment by xingpingcn** [comment]
> 在控制台有没有抛错？

[chat.xingpingcn.top-1716164970665.log](https://github.com/lobehub/lobe-chat/files/15371358/chat.xingpingcn.top-1716164970665.log)

且在network查看该post请求的状态码是200，日志显示的是”The user aborted a request“

另外使用的是edge，已经清除浏览器缓存。且初次打开的时候不断加载了这些js，非常卡

![image](https://github.com/lobehub/lobe-chat/assets/117259690/ad6831b2-3798-45dc-bc73-a8c473dbbfb7)

**Comment by arvinxx** [comment]
怎么样可以复现这个 user abort？

[a comment by an automated account is not shown]

**Comment by xingpingcn** [comment]
> 怎么样可以复现这个 user abort？

我让Gemini写一篇1k字以上的关于ai的文章，基本上输出的字体一多就容易触发，但是不是100%触发

[a comment by an automated account is not shown]

**Comment by xingpingcn** [comment]
> preview 用的是 vercel

因为是走的本地代理，而不是netlify服务器，现在临时的解决方法就是手动填写api和proxy了，但是这样子在环境变量填写的东西就失去意义了。解决的方法我想也很简单，只需要提供一个不用手动填写api和proxy就能走本地代理的按钮就行，api和proxy从netlify服务器环境变量中获取就可以

[a comment by an automated account is not shown]

**Comment by jimmyshjj** [comment]
> api和proxy从netlify服务器环境变量中获取就可以

API key不应发送到前端

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
> > preview 用的是 vercel
> 
> 
> 
> 因为是走的本地代理，而不是netlify服务器，现在临时的解决方法就是手动填写api和proxy了，但是这样子在环境变量填写的东西就失去意义了。解决的方法我想也很简单，只需要提供一个不用手动填写api和proxy就能走本地代理的按钮就行，api和proxy从netlify服务器环境变量中获取就可以

如果前端调api key可以，但后端调用出现截断，那你可能要检查下是不是 netlify 的serverless 时间超时导致的了。比如像 vercel 的nodejs runtime 默认只支持 10s 的调用时长，超过10s就会自动销毁，表现出来就是生成一部分内容以后就掐断了。

[a comment by an automated account is not shown]

**Comment by xingpingcn** [comment]
> > > preview 用的是 vercel
> > 
> > 
> > 因为是走的本地代理，而不是netlify服务器，现在临时的解决方法就是手动填写api和proxy了，但是这样子在环境变量填写的东西就失去意义了。解决的方法我想也很简单，只需要提供一个不用手动填写api和proxy就能走本地代理的按钮就行，api和proxy从netlify服务器环境变量中获取就可以
> 
> 如果前端调api key可以，但后端调用出现截断，那你可能要检查下是不是 netlify 的serverless 时间超时导致的了。比如像 vercel 的nodejs runtime 默认只支持 10s 的调用时长，超过10s就会自动销毁，表现出来就是生成一部分内容以后就掐断了。

Normally, functions are synchronous with a maximum execution time of 10 seconds. But, adding -background to the end of the filename transforms any function into an asynchronous function that can run in the background with up to 15 minutes of runtime.

![image](https://github.com/lobehub/lobe-chat/assets/117259690/702ea268-7811-48a5-8ddd-3a2aec8fc244)

netlify可以提升到26s，但是要钱。不知道能不能按照它所说的改成异步来增加超时限制

**Comment by arvinxx** [comment]
> netlify可以提升到26s，但是要钱

那你换成 vercel 部署就不会有这个问题了呗。

> 不知道能不能按照它所说的改成异步来增加超时限制

不可能，所有 ai 的 SSE 请求都是在一个 http 请求完成的，改成异步没有任何收益。

[a comment by an automated account is not shown]

**Comment by RubuJam** [comment]
你在`vercel`平台部署时，如果频繁遇到截断，可以试一试这个方法，应该能大幅度缓解你所遇到的情况https://github.com/lobehub/lobe-chat/discussions/2639#discussioncomment-9544307
